File: hw/pastor/u1/hw06Dependencies/chickenInJavaScript/Chicken.model.js
class Chicken {
    constructor(id, name, color, age, isMolting) {
        this._id = id;
        this._name = name;
        this._color = color;
        this._age = age;
        this._isMolting = isMolting;
    }


    toString(){
        return `Chicken: \n id --> \t ${this._id} \n name --> \t ${this._name} \n color --> \t ${this._color} \n age --> \t ${this._age} \n isMolting --> \t ${this._isMolting}`;
    }


    getId() {
        return this._id;
    }

    getName() {
        return this._name;
    }

    getColor() {
        return this._color;
    }   

    getAge() {
        return this._age;
    }   

    getIsMolting() {
        return this._isMolting;
    }   

    setId(id) {
        this._id = id;
    }

    setName(name) {
        this._name = name;
    }   

    setColor(color) {
        this._color = color;
    }

    setAge(age) {
        this._age = age;
    }   

    setIsMolting(isMolting) {
        this._isMolting = isMolting;
    }
}

module.exports = Chicken;

File: hw/medina/u1/hw06ChickenFarmSimulator/chicken - Python/Chicken_model.py
class Chicken:
    def __init__(self, id: int, name: str, color: str, age: int, is_molting: bool):
        self._id = id
        self._name = name
        self._color = color
        self._age = age
        self._is_molting = is_molting

    def set_id(self, id: int):
        self._id = id

    def get_id(self) -> int:
        return self._id

    def set_name(self, name: str):
        self._name = name

    def get_name(self) -> str:
        return self._name

    def set_color(self, color: str):
        self._color = color

    def get_color(self) -> str:
        return self._color

    def set_age(self, age: int):
        self._age = age

    def get_age(self) -> int:
        return self._age

    def set_is_molting(self, is_molting: bool):
        self._is_molting = is_molting

    def get_is_molting(self) -> bool:
        return self._is_molting

    def __str__(self) -> str:
        return f"Chicken: id={self._id}, name={self._name}, color={self._color}, age={self._age}, is_molting={self._is_molting}"
